app/connectors/AgentPermissionsConnector.scala

```diff
@@ -177,9 +180,10 @@ class AgentPermissionsConnectorImpl @Inject() (val http: HttpClient)(implicit
   }
 
   def optIn(arn: Arn, lang: Option[String])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Done] = {
-    val url = s"$agentPermissionsUrl/arn/${arn.value}/optin" + lang.fold("")("?lang=" + _)
+    val urlQuery: String = lang.fold("")(x => s"?lang=$x")
+    val url: URL = url"$agentPermissionsUrl/arn/${arn.value}/optin$urlQuery"
```

Comment:
Can you confirm this url works right, I believe appending query params like this can utf encode the ? and & signs, breaking the url